sklearn/preprocessing/_data.py

```diff
@@ -1560,8 +1560,12 @@ def transform(self, X):
                 to_stack.append(np.ones(shape=(n_samples, 1), dtype=X.dtype))
             to_stack.append(X)
             for deg in range(2, self.degree+1):
-                Xp_next = _csr_polynomial_expansion(X.data, X.indices,
-                                                    X.indptr, X.shape[1],
+                # use np.int64 for index datatype to prevent overflow
+                # in case X has a large dimension
+                Xp_next = _csr_polynomial_expansion(X.data,
+                                                    X.indices.astype(np.int64),
```

Comment:
What kind of error do you get when using the fused type without casting to `np.int64`?

Reply:
in my case the overflow error still occurs if no casting was used, even when the type is a fused type.

Reply:
A self-contained example code would make it easier to reproduce your issue. May you expand on the code snippet in https://github.com/scikit-learn/scikit-learn/issues/16803#issue-589922066 with code to create `x` as well?

Reply:
@thomasjpfan I have updated the code snippet in #16803 to be self-contained and can reproduce this issue.